sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/WriteDocumentsService.java

```diff
@@ -0,0 +1,435 @@
+package io.stargate.sgv2.docsapi.service.write;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCode;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCodeRuntimeException;
+import io.stargate.sgv2.docsapi.api.v2.model.dto.DocumentResponseWrapper;
+import io.stargate.sgv2.docsapi.api.v2.model.dto.MultiDocsResponse;
+import io.stargate.sgv2.docsapi.config.DocumentConfig;
+import io.stargate.sgv2.docsapi.service.ExecutionContext;
+import io.stargate.sgv2.docsapi.service.JsonDocumentShredder;
+import io.stargate.sgv2.docsapi.service.JsonShreddedRow;
+import io.stargate.sgv2.docsapi.service.query.ReadBridgeService;
+import io.stargate.sgv2.docsapi.service.schema.JsonSchemaManager;
+import io.stargate.sgv2.docsapi.service.schema.TableManager;
+import io.stargate.sgv2.docsapi.service.schema.qualifier.Authorized;
+import io.stargate.sgv2.docsapi.service.util.DocsApiUtils;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class WriteDocumentsService {
+  @Inject ObjectMapper objectMapper;
+
+  @Inject JsonSchemaManager jsonSchemaManager;
+
+  @Inject @Authorized TableManager tableManager;
```

Comment:
So there was actually a reason why I had done this - it was out of frustration, and then I forgot to fix it properly

Because TableManager has a sub-class AuthorizedTableManager, @Inject works fine in tests, but @InjectMock does NOT. The mocking returns an error because there are two classes that match the injection (both Authorized and "not authorized" versions of TableManager are both instances of TableManager). Attempting to add `@Typed` annotation to the AuthorizedTableManager causes everything else to fail - the dependency injection stops working...

So what I decided to do to fix this is to make a new class that is a trivial subclass of TableManager, called `NoAuthTableManager`. This has no implementation, but makes it so that you can inject it as a mock explicitly in tests, and there will be no ambiguity in the dependency injection.

Reply:
Are u sure? In tests we inject normally without authorized annotation, and ut works. You can try with @Default. Eaither way we should not have a sub-class, there is a solution without having this extra class for sure, it s standard injection stuff.